chainstate/types/src/block_index.rs

```diff
@@ -39,6 +40,22 @@ pub struct BlockIndex {
     status: BlockStatus,
 }
 
+impl Idable for BlockIndex {
+    type Tag = Block;
+
+    fn get_id(&self) -> Id<Block> {
+        *self.block_id()
+    }
+}
```

Comment:
BlockIndex is not idable. 

Why is this necessary?

Reply:
> BlockIndex is not idable

I've removed this.